### PR TITLE
allow some sui tool commands to take a fullnode-rpc arg and use the latest committee

### PR DESCRIPTION
**Pull request by longbowlu**
## Description 

As title. Also remove some stale implementations and use concise keys.

Example:
```
./sui-tool fetch-object --id 0xa577f89fcadd5805f2a9ae0b12670d3051d938e34c702484d2a5b15dd92bc057   --fullnode-rpc https://rpc.mainnet.sui.io:443
```

## Test Plan 

locally tested sui-tool

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes
Allow some sui tool commands to take a fullnode-rpc arg and use the latest committee. Also remove some stale implementations and use concise keys.

### CODE REVIEW

**Review by arun-koshy — APPROVED** [review]

File: crates/sui-tool/src/lib.rs [hunk 1]
```diff
@@ -377,59 +405,43 @@ pub async fn get_transaction_block(
...
                 res.3
             )?;
         }
-        writeln!(&mut s)?;
+        writeln!(&mut s, "{:<100}\n", "-".repeat(100))?;
     }
     Ok(s)
 }
 
+// Keep the return types a vector in case we need support for lamport versions in the near future
```
Comment (mwtian) [inline crates/sui-tool/src/lib.rs:420]: s/types/type ?

File: crates/sui-tool/src/lib.rs [hunk 2]
```diff
@@ -377,59 +405,43 @@ pub async fn get_transaction_block(
...
-                    }
-                },
-            })
-            .await
-            .map_err(anyhow::Error::from);
-        let elapsed = start.elapsed().as_secs_f64();
-
-        let resp_version = resp.as_ref().ok().map(|r| r.object.version().value());
-        ret.push((resp_version.map(SequenceNumber::from), resp, elapsed));
+    let version = start_version;
```
Comment (mwtian) [inline crates/sui-tool/src/lib.rs:427]: Is it necessary to set `start_version` to `version`?

File: crates/sui-tool/src/commands.rs [hunk 3]
```diff
@@ -54,11 +54,14 @@ pub enum ToolCommand {
...
+        // At least one of genesis or fullnode_rpc must be provided
         #[clap(long = "genesis")]
-        genesis: PathBuf,
+        genesis: Option<PathBuf>,
 
-        #[clap(long = "history", help = "show full history of object")]
-        history: bool,
+        // At least one of genesis or fullnode_rpc must be provided
+        // RPC address to provide the up-to-date committee info
+        #[clap(long = "fullnode-rpc")]
```
Comment (mwtian) [inline crates/sui-tool/src/commands.rs:63]: Maybe useful to provide an example. Also is it better to name this `fullnode-addr`?

**Review by mwtian — APPROVED** [review]

File: crates/sui-tool/src/commands.rs (hunk 3 above)
Comment (longbowlu) [reply crates/sui-tool/src/commands.rs]: renamed to `fullnode-rpc-url` to be consistent with other cli tools

**Pull request merged**